src/main/java/com/iexec/blockchain/broker/BrokerService.java

```diff
@@ -115,41 +136,52 @@ private Optional<String> fireMatchOrders(BrokerOrder brokerOrder) {
         return Optional.empty();
     }
 
-
-    private boolean hasRequesterAcceptedPrices(
+    boolean hasRequesterAcceptedPrices(
             RequestOrder requestOrder,
             BigInteger appPrice,
             BigInteger workerpoolPrice,
-            BigInteger datasetPrice
+            BigInteger datasetPrice,
+            boolean withDataset
     ) {
-        if (requestOrder == null || requestOrder.getWorkerpoolmaxprice() == null
-                || requestOrder.getAppmaxprice() == null
-                || appPrice == null || workerpoolPrice == null) {
-            log.error("Failed to check hasRequesterAcceptedPrices (null requestOrder)");
-            return false;
-        }
         boolean isAppPriceAccepted = requestOrder.getAppmaxprice().longValue() >= appPrice.longValue();
         boolean isPoolPriceAccepted = requestOrder.getWorkerpoolmaxprice().longValue() >= workerpoolPrice.longValue();
-        boolean isDatasetPriceAccepted = requestOrder.getDatasetmaxprice().longValue() >= datasetPrice.longValue();
-        boolean isAccepted = isAppPriceAccepted && isPoolPriceAccepted && isDatasetPriceAccepted;
+        boolean isAccepted = isAppPriceAccepted && isPoolPriceAccepted;
+        String messageDetails = MessageFormat.format("[isAppPriceAccepted:{0}, isPoolPriceAccepted:{1}]",
+                isAppPriceAccepted, isPoolPriceAccepted);
+        if (withDataset) {
+            boolean isDatasetPriceAccepted = requestOrder.getDatasetmaxprice().longValue() >= datasetPrice.longValue();
+            isAccepted = isAccepted && isDatasetPriceAccepted;
+            messageDetails = MessageFormat.format("[isAppPriceAccepted:{0}, isPoolPriceAccepted:{1}, isDatasetPriceAccepted:{2}]",
+                    isAppPriceAccepted, isPoolPriceAccepted, isDatasetPriceAccepted
+            );
+        }
         if (!isAccepted) {
-            log.error("Prices not accepted (too expensive) [isAppPriceAccepted:{}, " +
-                    "isPoolPriceAccepted:{}]", isAppPriceAccepted, isPoolPriceAccepted);
+            log.error("Prices not accepted (too expensive) {}", messageDetails);
         }
         return isAccepted;
     }
 
-    private boolean hasRequesterDepositedEnough(RequestOrder requestOrder, long deposit) {
-        if (requestOrder == null || requestOrder.getWorkerpoolmaxprice() == null
-                || requestOrder.getAppmaxprice() == null) {
-            log.error("Failed to check hasRequesterDepositedEnough (null requestOrder)");
-            return false;
+    boolean hasRequesterDepositedEnough(RequestOrder requestOrder, long deposit, boolean withDataset) {
```

Comment:
```suggestion
    //TODO: Check should verify deposited >= appOrder.price + appOrder.price + data.price
    // (instead of deposited >= reqOrderAppMaxPrice + reqOrderPoolMaxPrice + reqOrderDataMaxPrice)
    boolean hasRequesterDepositedEnough(RequestOrder requestOrder, long deposit, boolean withDataset) {
```

Reply:
see 61eef35